### ISSUE TITLE
Language selection sometimes changes automatically to English

### ISSUE THREAD

**Issue by IgnacioMilani** [LAION-AI/Open-Assistant#1411]
1- Select language Spanish (Español)
2- Enter to Messages (Mensajes)
3- Click on triple dots of a message and Open on new tab (Abrir en una pestaña nueva). 

![image](https://user-images.githubusercontent.com/32331378/218004370-072115bf-85b1-4fd3-a914-68e2a0d4e14a.png)

4- It automatically changes language to English.
![image](https://user-images.githubusercontent.com/32331378/218004428-ab27c561-8557-421b-85e9-b5c55f737dc9.png)

**Comment by IgnacioMilani** [comment]
Inspecting the source code I found out the button has this link:
`<a href="/messages/215c07b2-b764-4713-8ed3-a3ae88065c51"`
When it should be:
`<a href="/es/messages/215c07b2-b764-4713-8ed3-a3ae88065c51"`

This issue persists in all languages.

**Comment by SiddheshBangar** [comment]
Hello, I can try out this one. 
I am new to open source so would like to contribute to it.

**Comment by GeroVanMi** [comment]
As far as I see we would need to change a line in `website/src/components/Messages/MessageTableEntry.tsx`.

There is the following line: 
```html
        <MenuItem as="a" href={`/messages/${id}`} target="_blank" icon={<MessageSquare />}>
          {t("open_new_tab_action")}
        </MenuItem>
```
Here we'd somehow need to add the current language to the `href={'/messages/${CURRENT_LANGUAGE}/${id}'}` part, but I don't know how to retrieve the current language.

**Comment by andreaskoepf** [comment]
> Hello, I can try out this one. I am new to open source so would like to contribute to it.

I suggest that you join the OA discord server. Probably the web-team can tell you how to get the current language code.

**Comment by GeroVanMi** [comment]
I found a solution that works but (in my opinion) might not be very clean. (I haven't asked the web-team yet) 

```typescript
  // Existing code for refrence (around line 177)
  const { trigger: deleteMessage } = useSWRMutation(`/api/admin/delete_message/${id}`, del);
  // Newly added code
  const [cookies,] = useCookies(["NEXT_LOCALE"]);
```

```html
<MenuItem as="a" href={`/${cookies['NEXT_LOCALE']}/messages/${id}`} target="_blank" icon={<MessageSquare />}>
    {t("open_new_tab_action")}
</MenuItem>
```

**Comment by mehdi-zibout** [comment]
@GeroVanMi check this https://nextjs.org/docs/advanced-features/i18n-routing.

**Comment by SiddheshBangar** [comment]
> > Hello, I can try out this one. I am new to open source so would like to contribute to it.
> 
> I suggest that you join the OA discord server. Probably the web-team can tell you how to get the current language code.

Hey thanks @andreaskoepf for assigning me. 
Can you share me the discord server link?

**Comment by JohannesGaessler** [comment]
Probably the same issue: https://github.com/LAION-AI/Open-Assistant/issues/1365

Steps to reproduce: https://github.com/LAION-AI/Open-Assistant/issues/1365#issuecomment-1425746693

**Comment by notmd** [comment]
For anyone want to take this, changing `a` to `NextLink` should work. Next will handle the locale for you.
See [https://stackoverflow.com/questions/65632698/how-to-open-a-link-in-a-new-tab-in-nextjs](https://stackoverflow.com/questions/65632698/how-to-open-a-link-in-a-new-tab-in-nextjs)

**Comment by JohannesGaessler** [comment]
There may be more than one bug that could be causing this. Consider the following screenshots:

![Screenshot_20230214_130045](https://user-images.githubusercontent.com/18492268/218732757-e4756ee8-308b-4a0c-99e8-dc2e02adbaad.png)
![Screenshot_20230214_130108](https://user-images.githubusercontent.com/18492268/218732787-f9bafac6-fbf9-402b-9956-23378fd1f248.png)

I am doing a task for English but I am being shown a German tree. When labeling the assistant reply (both below the tree and when clicking ... -> Label) I can label it as "Not English". However, when labeling the prompt I can label it as "Not German". I did not open any links to a German tree like I did in https://github.com/LAION-AI/Open-Assistant/issues/1365#issuecomment-1425746693 .

Link: https://open-assistant.io/messages/0b31aa89-fb97-4f5d-916c-5acff94c1435
